lib/Doctrine/ORM/Query/Parser.php

```diff
@@ -2367,56 +2367,58 @@ public function SimpleSelectExpression()
     {
         $peek = $this->lexer->glimpse();
 
-        switch ($this->lexer->lookahead['type']) {
-            case Lexer::T_IDENTIFIER:
-                switch (true) {
-                    case $peek['type'] === Lexer::T_DOT:
-                        $expression = $this->StateFieldPathExpression();
+        if ($this->lexer->lookahead !== null) {
+            switch ($this->lexer->lookahead['type']) {
```

Comment:
```suggestion
        switch ($this->lexer->lookahead['type'] ?? null) {
```

Let's keep the diff small here.

Reply:
Done, I had to add a check to the line 2390